lib/index.js

```diff
@@ -73,7 +73,6 @@ const internals = {
 exports.wrap = function (error, statusCode, message) {
 
     Hoek.assert(error instanceof Error, 'Cannot wrap non-Error object');
-    Hoek.assert(!error.isBoom || (!statusCode && !message), 'Cannot provide statusCode or message with boom error');
```

Comment:
As far as I understood the goal of the function is only to wrap an error with Boom information in case it is not already a Boom object. IMO considering that it should not do anything in the case it is already a Boom error. It could be used as a global error wrapper in case it is not an error we threw ourselves with Boom.